src/main/java/org/hl7/davinci/priorauth/ClaimResponseFactory.java

```diff
@@ -239,4 +242,59 @@ else if (atleastOneDenied && !atleastOneGranted)
         return itemComponent;
     }
 
+    public static Bundle handleClaimInquiry(Bundle claimInquiryBundle) {
+        String id = FhirUtils.getIdFromResource(claimInquiryBundle);// this should be inquiry but if it's not
+        String patient = FhirUtils.getPatientIdentifierFromBundle(claimInquiryBundle);
+        Claim claim = (Claim) App.getDB().read(Table.CLAIM, Collections.singletonMap("patient", patient));
+        List<Claim.ItemComponent> queriedItems = claim.getItem();
```

Comment:
This filtering is going to need to be a little bit more complex. Let me know if you need help with that.